File: disk_monitor.py
import shutil

# Path to check
path = "/"  # On Windows, you can use "C:\\"

total, used, free = shutil.disk_usage(path)
percent_free = (free / total) * 100

print(f"Disk free: {percent_free:.2f}%")

if percent_free < 10:
    print("[ALERT] Disk space is below 10%!")
else:
    print("[OK] Disk space is sufficient.")

File: Go disk_monitor/disk_monitor.go
// package main

// import (
// 	"fmt"

// 	"github.com/shirou/gopsutil/v3/disk"
// )

// func main() {
// 	path := `C:\` // Use "/" on Linux

// 	usage, err := disk.Usage(path)
// 	if err != nil {
// 		fmt.Println("[ERROR] Cannot get disk stats:", err)
// 		return
// 	}

// 	percentFree := (float64(usage.Free) / float64(usage.Total)) * 100
// 	fmt.Printf("Disk free: %.2f%%\n", percentFree)

// 	if percentFree < 10 {
// 		fmt.Println("[ALERT] Disk space is below 10%!")
// 	} else {
// 		fmt.Println("[OK] Disk space is sufficient.")
// 	}
// }
